internal/server/providers/gh/gh_e2e_test.go

```
@@ -0,0 +1,699 @@
+	raw := postQueryRaw(t, url, query)
+	var out graphqlResponse
+	if err := json.Unmarshal(raw, &out); err != nil {
+		t.Fatalf("decode %q: %v", raw, err)
+	}
+	return out
+}
+
+func postQueryRaw(t *testing.T, url, query string) []byte {
+	t.Helper()
+	body, _ := json.Marshal(map[string]string{"query": query})
+	req, err := http.NewRequest(http.MethodPost, url+"/graphql", strings.NewReader(string(body)))
```

golangci-lint (2.12.1)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)